syncstorage/storage/__init__.py

```diff
@@ -307,6 +313,63 @@ def delete_items(self, userid, collection, items):
             CollectionNotFoundError: the user has no such collection.
         """
 
+    @abc.abstractmethod
+    def create_transaction(self, userid, collection):
+        """Creates a transaction for multi-POST batch uploads.
+
+        Args:
+            userid: integer identifying the user in the storage.
+            collection: name of the collection.
+
+        Returns:
+            The new transaction's ID as a BigInteger timestamp.
+        """
+
+    @abc.abstractmethod
+    def append_items_to_transaction(self, batchid, userid, collection, items):
+        """Creates or updates multiple items from a multi-POST batch to a
+        transaction.
+
+        Args:
+            batchid: big integer batch identifier for this transaction
```

Comment:
The naming confusion between "batchid" and "transaction" really stands out in the code here; if the name of the thing is to be "transaction" then lets make this "trnid" or "transactionid" or something all throughout the code and the API.